ocaml/xapi/context.ml

```diff
@@ -425,13 +424,12 @@ let get_user_agent context =
 let with_tracing context name f =
   let open Tracing in
   let parent = context.tracing in
-  let tracer = get_tracer ~name in
-  match Tracer.start ~tracer ~name ~parent () with
-  | Ok span ->
+  let tracing = start_tracing_helper (fun _ -> parent) name in
+  match tracing with
```

Comment:
It's not necessary to bind the result of `start_tracing_helper (fun _ -> parent) name` to the `tracing` variable. `match start_tracing_helper (fun _ -> parent) name with` is ok.

Reply:
This change should fix it: https://github.com/xapi-project/xen-api/pull/5267/commits/a1c796623ad4bcf6ed1b5fc3696d4a350f56c79b .